extension/chrome/settings/setup/setup-create-key.ts

```diff
@@ -37,8 +36,19 @@ export class SetupCreateKeyModule {
       if (this.view.clientConfiguration.canBackupKeys()) {
         await this.view.submitPublicKeys(opts);
         const action = $('#step_2a_manual_create .input_backup_inbox').prop('checked') ? 'setup_automatic' : 'setup_manual';
-        // only finalize after backup is done. backup.htm will redirect back to this page with ?action=finalize
-        window.location.href = Url.create('modules/backup.htm', { action, acctEmail: this.view.acctEmail, idToken: this.view.idToken, id: keyIdentity.id, type: keyIdentity.family });
+        // only finalize after backup is done.
+        $('#step_2a_manual_create').hide();
+        await this.view.backupUi.initialize({
+          acctEmail: this.view.acctEmail,
+          action,
+          keyIdentityId: keyIdentity.id,
+          keyIdentityFamily: keyIdentity.family,
```

Comment:
we need to check if family is `openpgp` and define `keyIdentity` in 2 places.
Rather than having duplicated code, I passed keyIdentityId and family.
Let me know your thoughts.

![image](https://user-images.githubusercontent.com/77344191/178739926-950e4b50-61b5-405b-89da-89d8d11fa513.png)

Reply:
I would think, whenever we have a type that contains several properties, and we need to pass most/all of those properties, then it makes sense to pass the type and not individual parts.

So if you already have an existing type `KeyIdentity` with paremeters `id` and `family` and you plan on passing both, might as well just pass `keyIdentity: KeyIdentity` instead of passing them individually.

Reply:
ok